Source/WebCore/accessibility/mac/AXObjectCacheMac.mm

```diff
@@ -399,12 +399,33 @@ static void exerciseIsIgnored(AccessibilityObject& object)
 
 void AXObjectCache::onDocumentRenderTreeCreation(const Document& document)
 {
-    if (m_sortedIDListsInitialized) {
-        // We only need to do this work if the sorted ID list has already been initialized.
-        m_deferredDocumentAddedList.add(document);
-        if (!m_performCacheUpdateTimer.isActive())
-            m_performCacheUpdateTimer.startOneShot(0_s);
-    }
+    if (!m_sortedIDListsInitialized)
+        return;
+    // We only need to do this work if the sorted ID list has already been initialized.
+
+    RefPtr object = getOrCreate(document.renderView());
+    if (!object || !object->isWebArea())
+        return;
+    queueUnsortedObject(object.releaseNonNull(), PreSortedObjectType::WebArea);
+}
+
+void AXObjectCache::deferSortForLiveRegion(Ref<AccessibilityObject>&& object)
+{
+    if (!m_sortedIDListsInitialized)
+        return;
+    // We only need to do this work if the sorted ID list has already been initialized.
+    queueUnsortedObject(WTFMove(object), PreSortedObjectType::LiveRegion);
+}
+
+void AXObjectCache::queueUnsortedObject(Ref<AccessibilityObject>&& object, PreSortedObjectType type)
+{
+    auto unsortedObjectListIterator = m_deferredUnsortedObjects.ensure(type, [&] {
+        return Vector<Ref<AccessibilityObject>>();
+    }).iterator;
+    unsortedObjectListIterator->value.appendIfNotContains(WTFMove(object));
+
+    if (!m_performCacheUpdateTimer.isActive() && !m_performingDeferredCacheUpdate)
```

Comment:
Could we have a schedulePerformCacheUpdateTimerIfNeeded() helper that we could use throughout the code?